### PR TITLE
Add periodic report

### PR DESCRIPTION
**Pull request by ChiragKumar9**
Self scheduling periodic report that implements a given closure with a given periodicity.

### CODE REVIEW

File: src/report.rs [hunk 1]
```diff
@@ -138,6 +166,26 @@ impl ContextReportExt for Context {
         let data_container = self.get_data_container_mut(ReportPlugin);
         &mut data_container.config
     }
+
+    /// Adds a periodic report to the context. The report will be output every `report_period` time
+    /// and will print out the output from the `report_output_fcn` function.
+    fn add_periodic_report<T: Report + 'static, C: Fn(&Context) -> T + 'static>(
```
Comment (k88hudson-cfa) [inline src/report.rs:172]: I think this would actually be better as a more generic `add_periodic_plan_with_priority`: would you mind moving it there instead? you could still have a test/example for adding the report and then scheduling it. 

Also, I think then C can just be generic then (like other plans).

LGTM with that change

**Review by k88hudson-cfa — CHANGES_REQUESTED** [review]

File: src/context.rs [hunk 2]
```diff
@@ -158,6 +158,50 @@ impl Context {
...
+    /// if needed.
+    /// # Panics
+    ///
+    /// Panics if time is negative, infinite, or NaN.
+    pub fn add_periodic_plan_with_phase(
+        &mut self,
+        period: f64,
+        callback: impl Fn(&mut Context) + 'static,
+        phase: ExecutionPhase,
+    ) {
```
Comment (ChiragKumar9) [inline src/context.rs:192]: My claim is that periodic plans should not return `Id` because that `Id` has no meaning after the first period's plan has run.

File: src/report.rs [hunk 3]
```diff
@@ -213,12 +213,8 @@ mod test {
     #[test]
     fn add_report_no_dir() {
```
Comment (ChiragKumar9) [inline src/report.rs:214]: I rewrote this test to be true to its name (previously it was just a repeat of the test above). I realize I should mayb pull this out and put in its own PR?
